src/highlevel.jl

```diff
@@ -61,21 +60,20 @@ Fields:
 """
 mutable struct Device
     ptr::Ptr{SoapySDRDevice}
-    function Device(dev_ptr)
+    function Device(args)
```

Comment:
We should also support a `Device(f::Function, args)` wrapper that explicitly closes the device after `f()` is done, so we can do `do`-block style resource management:

```
Device(args) do dev
    ...
end
```